packages/core/src/browser/tree/tree-expansion.ts

```diff
@@ -25,7 +25,7 @@ export interface ITreeExpansionService extends Disposable {
      *
      * Return true if a node has been expanded; otherwise false.
      */
-    expandNode(node: Readonly<IExpandableTreeNode>): boolean;
+    expandNode(node: Readonly<IExpandableTreeNode>): Promise<boolean>;
```

Comment:
please check and adapt existing clients

Reply:
ok

Reply:
Is it mandatory to change everything to `Promise`? Isn't it strange, half of the methods sync and the other half is async in on the model?

Reply:
It is a minimal change to implement the task. I find it is cleaner than listen to `onDidRefresh` events.

Whether It could be useful to promisify remaining `select` methods I don't know.

Reply:
If it works (and will work) well with Phosphor, I do not have any objections.

Reply:
it should be fine.